app/javascript/scavenger_hunt/components/MicrosoftLoginButton/MicrosoftLoginButton.jsx

```diff
@@ -1,13 +1,40 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 import MicrosoftLogin from "react-microsoft-login";
+import { useCreateHunterMutation } from "../../data/mutations"
  
 const MicrosoftLoginButton = (props) => {
+  const [ user, setUser ] = useState({ name: null, email: null });
+  const [ errors , setErrors ] = useState(null);
+
+  const [createHunter, { loading }] = useCreateHunterMutation();
+
+  useEffect(() => {
+    if (!loading && user.name != null && user.email != null) {
+      createHunter({
+        variables: {
+          input: {
+            "email": user.email,
+            "name": user.name,
+          }
+        }
+      }).then((res) => {
+        console.log(res.data.createHunter.errors);
```

Comment:
This is temporary, I am working on creating an ErrorProvider which can be wrapped around App component for it to catch any error/notice from any page